### PR TITLE
Refactor the wagtail sitemap generator.

### PR DESCRIPTION
**Pull request by mvantellingen**
This new version is based on the Django sitemaps and therefore can be
easily mixed with other sitemaps in one index. It should be completly
backwards compatible.

This also implements support for pagination (see #1698)

Co-authored with @mikedingjan

### CODE REVIEW

**Comment by mvantellingen** [comment]
This PR is missing the documentation update. But perhaps we should first check if everyone agrees with this change before we invest time into that :-)

**Comment by tomdyson** [comment]
Thanks, @mvantellingen and @mikedingjan. LGTM. Go ahead with the docs :)

File: docs/reference/contrib/sitemaps.rst [hunk 1]
```diff
@@ -3,24 +3,42 @@
...
+
+    As of Wagtail 1.10 the Django contrib sitemap app is used to generate
+    sitemaps.  However since Wagtail requires the Site instance to be available
+    during the sitemap generation you will have to use the views from the
+    ``wagtail.contrib.wagtailsitemaps.views`` module instead of the views
+    provided by Django (``django.contrib.sitemaps.views``). 
+    
+    The usage of these views is otherwise identical, which means that
+    customisation and caching of the sitemaps are done using the default Django
+    patterns.  See for in-depth information the Django documentation.
```
Comment (tomdyson) [inline docs/reference/contrib/sitemaps.rst]: -> "See the Django documentation for in-depth information"

File: docs/reference/contrib/sitemaps.rst [hunk 2]
```diff
@@ -84,10 +109,14 @@ Each dictionary can contain the following:
...
-Cache
------
+Serving multiple sitemaps
+~~~~~~~~~~~~~~~~~~~~~~~~~
 
-By default, sitemaps are cached for 100 minutes. You can change this by setting ``WAGTAILSITEMAPS_CACHE_TIMEOUT`` in your Django settings to the number of seconds you would like the cache to last for.
+If you want to support the sitemap indexes from Django then you will need to
+use the index view from ``wagtail.contrib.sitemaps.views`` instead of the
+index view from ``django.contrib.sitemaps.views``.  Please see for further
+details the Django documentation.
```
Comment (tomdyson) [inline docs/reference/contrib/sitemaps.rst]: -> Please see the Django documentation for further details.

**Pull request merged**